streams/examples/src/main/java/org/apache/kafka/streams/examples/temperature/TemperatureDemo.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.examples.temperature;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.utils.Exit;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.KStreamBuilder;
+import org.apache.kafka.streams.kstream.KeyValueMapper;
+import org.apache.kafka.streams.kstream.Predicate;
+import org.apache.kafka.streams.kstream.Reducer;
+import org.apache.kafka.streams.kstream.TimeWindows;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.kstream.internals.WindowedDeserializer;
+import org.apache.kafka.streams.kstream.internals.WindowedSerializer;
+
+import java.util.Properties;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * Demonstrates, using the high-level KStream DSL, how to implement an IoT demo application
+ * which ingests temperature value processing the maximum value in the latest 5 seconds sending
+ * a new message if it exceeds 20
+ *
+ * In this example, the input stream reads from a topic named "temperature", where the values of messages
+ * represent temperature values; using a 5 seconds "tumbling" window, the maximum value is processed and
+ * sent to a topic named "max" if it exceeds 20
+ *
+ * Before running this example you must create the input topic and the output topic (e.g. via
```

Comment:
I'd suggest to include the actual, full CLI commands to create the input topic ("temperature") and the output topic ("max").  We should make it as easy as possible for beginners to get started -- some user might end up playing with Kafka for the first time because they want to run this specific Kafka Streams example.

Also, I'd suggest to somehow prefix the topic names so that it is a bit easier to tie them back to this example?  Like: "temperature" -> "iot-temperature", "max", "iot-temperature-max"?